### PR TITLE
Add option to change tag of uss file

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

Solves #2113 

Added support for changing file tags on USS files, the new react code features:

- conditional rendering for directories/files specifically for this new edit option 
- will show `NOT SUPPORTED` if the API for `getTag()` is not implemented by extender

New API in Zowe Explorer API was added: `getTag()`

## Screenshots

Supported Tag view on z/OSMF
Screenshot 2023-09-20 at 2 58 54 PM

Opening a Directory hides the tag view
Screenshot 2023-09-20 at 3 01 17 PM

If tag is not supported it will show `NOT SUPPORTED` and is also not editable or clickable
Screenshot 2023-09-20 at 3 02 17 PM

## Release Notes

Milestone: 2.12.0

Changelog: Added option for changing tags on USS files

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by rudyflores** [comment]
Keeping this in draft for the time being! currently writing some unit tests but once those are ready this should be ready to be reviewed!

**Comment by rudyflores** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=zowe_vscode-extension-for-zowe&pullRequest=2470)
> 
> [Bug](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=CODE_SMELL) [2 Code Smells](https://sonarcloud.io/project/issues?id=zowe_vscode-extension-for-zowe&pullRequest=2470&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=zowe_vscode-extension-for-zowe&pullRequest=2470) No Coverage information [](https://sonarcloud.io/component_measures?id=zowe_vscode-extension-for-zowe&pullRequest=2470&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=zowe_vscode-extension-for-zowe&pullRequest=2470&metric=new_duplicated_lines_density&view=list)

FYI the code smells are from tests mocking the `@zowe/cli` call for Utilities, this doesn't need to be addressed.

File: packages/zowe-explorer-api/src/profiles/ZoweExplorerZosmfApi.ts [hunk 1]
```diff
@@ -205,6 +212,14 @@ export class ZosmfUssApi extends ZosmfApiCommon implements ZoweExplorerApi.IUss
             apiResponse: result,
         };
     }
+
+    public async getTag(ussPath: string): Promise<string> {
+        const response = await zowe.Utilities.putUSSPayload(this.getSession(), ussPath, {
+            request: "chtag",
+            action: "list",
+        });
+        return JSON.parse(response.toString()).stdout[0].split(" ")[1] as string;
```
Comment (traeok) [inline packages/zowe-explorer-api/src/profiles/ZoweExplorerZosmfApi.ts:221]: This is a creative use of the `putUSSPayload` function, nice! 😋

**Review by traeok — APPROVED** [review]
This looks and works great! To test:
- I saved some ASCII data in an untagged file, then changed it to the EBCDIC tag (IBM-1047)
- Used "Pull from Mainframe", noticed the garbled characters once pulled down
- Changed the tag back to ISO-8859-1, used "Pull from Mainframe" again
- Verified that the ASCII data matched what I saved before changing tags

Thanks @rudyflores!

**Review by awharn — APPROVED** [review]

**Pull request merged**